### PR TITLE
chore(rust): log more errors as `tracing::Value`s

### PR DESCRIPTION
**Pull request by thomaseizinger**
Logging these as structured values gives us a better stacktrace in Sentry (assuming the errors themselves make proper use of defining an error-chain).

### CODE REVIEW

File: rust/bin-shared/src/network_changes/windows.rs [hunk 1]
```diff
@@ -374,6 +374,7 @@ impl Drop for Callback {
 
 mod async_dns {
     use anyhow::{Context as _, Result};
+    use firezone_logging::std_dyn_err;
```
Comment (thomaseizinger) [inline rust/bin-shared/src/network_changes/windows.rs]:
```suggestion
    use firezone_logging::anyhow_dyn_err;
```

File: rust/bin-shared/src/network_changes/windows.rs [hunk 2]
```diff
@@ -445,11 +446,17 @@ mod async_dns {
                 }
             }
 
-            if let Err(error) = listener_4.close() {
-                tracing::error!(?error, "Error while closing IPv4 DNS listener");
+            if let Err(e) = listener_4.close() {
+                tracing::error!(
+                    error = std_dyn_err(&e),
```
Comment (thomaseizinger) [inline rust/bin-shared/src/network_changes/windows.rs]:
```suggestion
                    error = anyhow_dyn_err(&e),
```

File: rust/bin-shared/src/network_changes/windows.rs [hunk 3]
```diff
@@ -445,11 +446,17 @@ mod async_dns {
...
+                    "Error while closing IPv4 DNS listener"
+                );
             }
-            if let Err(error) = listener_6.close() {
-                tracing::error!(?error, "Error while closing IPv6 DNS listener");
+            if let Err(e) = listener_6.close() {
+                tracing::error!(
+                    error = std_dyn_err(&e),
```
Comment (thomaseizinger) [inline rust/bin-shared/src/network_changes/windows.rs]:
```suggestion
                    error = anyhow_dyn_err(&e),
```

**Comment by thomaseizinger** [comment]
Finally fixed all CI errors. Ready for review @conectado @ReactorScram

File: rust/bin-shared/src/network_changes/windows.rs [hunk 4]
```diff
@@ -445,11 +446,17 @@ mod async_dns {
                 }
             }
 
-            if let Err(error) = listener_4.close() {
-                tracing::error!(?error, "Error while closing IPv4 DNS listener");
+            if let Err(e) = listener_4.close() {
+                tracing::error!(
+                    error = anyhow_dyn_err(&e),
+                    "Error while closing IPv4 DNS listener"
+                );
```
Comment (conectado) [inline rust/bin-shared/src/network_changes/windows.rs:453]: Just an idea for a future PR, should we make an `errror!` macro that allows us to do `error!(error, "...")`?

**Review by conectado — APPROVED** [review]
LGTM!

File: rust/bin-shared/src/network_changes/windows.rs (hunk 4 above)
Comment (thomaseizinger) [reply rust/bin-shared/src/network_changes/windows.rs:453]: We'd have to define a sigil that allows us to differentiate between anyhow and std-err. But maybe possible?

**Pull request merged**